api/router.js

```diff
@@ -0,0 +1,204 @@
+const router = require('express').Router();
+
+const db = require('./model.js');
+
+// '/api/projects' endpoints
+
+router.get('/projects', (req, res) => {
+    db.getProjects()
+    .then( project => {
+        res.status(200).json(project)
+    })
+    .catch(err => {
+        res.status(500).json(err)
+    })
+});
+
+router.get('/projects/:id', verifyProjectId, (req, res) => {
+    const id = req.params.id;
+
+    db.getProjectById(id)
+    .then( project => {
+        res.status(200).json(project)
+    })
+    .catch(err => {
+        res.status(500).json(err)
+    })
+});
+
+router.post('/projects', (req, res) => {
+    const { name, description } = req.body;
+    if( !name || !description ) {
+        res.status(400).json({ message: 'New projects require a name and description.' })
+    } else {
+        db.addProject(req.body)
+    .then(project => {
+        res.status(201).json(project)
+    })
+    .catch(err => {
+        res.status(500).json(err)
+    })
+    }
+});
+
+router.put('/projects/:id', verifyProjectId, (req, res) => {
+    const id = req.params.id;
+    const changes = req.body;
+    const { name, description } = req.body;
+    if( !name || !description ) {
+        res.status(400).json({ message: 'Projects require a name and description.' })
+    } else {
+    db.editProject(id, changes)
+    .then(updatedProject => {
+        res.status(201).json(updatedProject)
+    })
+    .catch(err => {
+        res.status(500).json(err)
+    })
+  }
+});
+
+router.delete('/projects/:id', verifyProjectId, (req, res) => {
+    const id = req.params.id;
+
+    db.deleteProject(id)
+    .then(deleted => {
+        res.status(200).json({ message: `${deleted} project deleted.` });
+    })
+    .catch(err => {
+        res.status(500).json(err)
+    })
+});
+
+
+// '/api/project/:id' endpoint
+
+router.get('/project/:id', verifyProjectId, (req, res) => {
+    const id = req.params.id;
+
+    db.getProjectWithActionsById(id)
```

Comment:
Everything in the route should probably be in the /projects/:id route on line 17